server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3919,4 +3961,34 @@ private SnapshotsInProgress createSnapshot(
             return res;
         }
     }
+
+    private record UpdateNodeIdsToRemoveTask() implements ClusterStateTaskListener {
```

Comment:
Nit: can we rename this a bit to keep consistency?
```suggestion
    private record UpdateNodeIdsForRemoveTask() implements ClusterStateTaskListener {
```

Reply:
sure, see bb3404b111a.